src/configuration.ts

```diff
@@ -1,28 +1,219 @@
+    }
+
+    switch (process.platform) {
+      case 'win32':
+        let programFiles = process.env['ProgramFiles'] || 'C:\\Program Files';
+        if (process.env['PROCESSOR_ARCHITEW6432'] === 'AMD64') {
+          programFiles = process.env['ProgramW6432'] || 'C:\\Program Files';
+        }
+        return path.join(programFiles, 'Puppet Labs', 'DevelopmentKit');
+      default:
+        return '/opt/puppetlabs/pdk';
+    }
+  }
+
+  get puppetDir(): string {
+    return path.join(this.puppetAgentDir, 'puppet');
+  }
+
+  get facterDir(): string {
+    return path.join(this.puppetAgentDir, 'facter');
+  }
+
+  get hieraDir(): string {
+    return path.join(this.puppetAgentDir, 'hiera');
+  }
+
+  get mcoDir(): string {
+    return path.join(this.puppetAgentDir, 'mcollective');
+  }
+
+  get rubydir(): string {
+    return path.join(this.puppetAgentDir, 'sys', 'ruby');
+  }
+
+  get rubylib(): string {
+    var p =
+      path.join(this.puppetDir, 'lib') +
+      this.pathEnvSeparator() +
+      path.join(this.facterDir, 'lib') +
+      this.pathEnvSeparator() +
+      path.join(this.hieraDir, 'lib') +
+      this.pathEnvSeparator() +
+      path.join(this.mcoDir, 'lib');
+
+    if (process.platform === 'win32') {
+      // Translate all slashes to / style to avoid puppet/ruby issue #11930
+      p = p.replace(/\\/g, '/');
+    }
 
-    this.host = config['languageserver']['address'];
-    this.port = config['languageserver']['port'];
-    this.timeout = config['languageserver']['timeout'];
-    this.enableFileCache = config['languageserver']['filecache']['enable'];
-    this.debugFilePath = config['languageserver']['debugFilePath'];
+    return p;
+  }
+
+  get sslCertDir(): string {
+    return path.join(this.puppetDir, 'ssl', 'certs');
+  }
+
+  get sslCertFile(): string {
+    return path.join(this.puppetDir, 'ssl', 'cert.pem');
+  }
+
+  get environmentPath(): string {
+    return (
+      path.join(this.puppetDir, 'bin') +
+      this.pathEnvSeparator() +
+      path.join(this.facterDir, 'bin') +
+      this.pathEnvSeparator() +
+      path.join(this.hieraDir, 'bin') +
+      this.pathEnvSeparator() +
+      path.join(this.mcoDir, 'bin') +
+      this.pathEnvSeparator() +
+      path.join(this.puppetAgentDir, 'bin') +
+      this.pathEnvSeparator() +
+      path.join(this.rubydir, 'bin') +
+      this.pathEnvSeparator() +
+      path.join(this.puppetAgentDir, 'sys', 'tools', 'bin') +
+      this.pathEnvSeparator()
+    );
+  }
+
+  get languageServerPath(): string {
+    return this.context.asAbsolutePath(path.join('vendor', 'languageserver', 'puppet-languageserver'));
+  }
+
+  get type(): ConnectionType {
+    if (this.host === '127.0.0.1' || this.host === 'localhost' || this.host === '') {
+      return ConnectionType.Local;
+    } else {
+      return ConnectionType.Remote;
+    }
+  }
 
-    this.puppetAgentDir = config['puppetAgentDir'];
+  get protocol(): ProtocolType {
+    switch (this.config['languageserver']['protocol']) {
```

Comment:
Here we specify the config as `puppet.languageserver.protocol` but in the [package.json](https://github.com/lingua-pupuli/puppet-vscode/pull/272/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R275) it is defined as `puppet.languageclient.protocol`. Both of these setting appear to be new and should map correctly - this currently causes connecting to a remote language server over TCP to fail.

Reply:
Fixed. Good catch, the default case statement catches the error and starts the STDIO server which is the only thing I cared about at the time. I think these setting names need a review, if a person familiar with them can't keep them straight :)